src/cbmc_starter_kit/template-for-repository/proofs/Makefile.common

```diff
@@ -788,60 +770,38 @@ $(LOGDIR)/coverage.xml: $(HARNESS_GOTO).goto
 	  --stderr-file $(LOGDIR)/coverage-err-log.txt \
 	  --description "$(PROOF_UID): calculating coverage"
 
-define VIEWER_CMD
-  $(VIEWER) \
-    --result $(LOGDIR)/result.txt \
-    --block $(LOGDIR)/coverage.xml \
-    --property $(LOGDIR)/property.xml \
-    --srcdir $(SRCDIR) \
-    --goto $(HARNESS_GOTO).goto \
-    --htmldir $(PROOFDIR)/html
-endef
-export VIEWER_CMD
-
-$(PROOFDIR)/html: $(LOGDIR)/result.txt $(LOGDIR)/property.xml $(LOGDIR)/coverage.xml
+$(PROOFDIR)/report: $(LOGDIR)/result.xml $(LOGDIR)/property.xml $(LOGDIR)/coverage.xml
 	$(LITANI) add-job \
-	  --command "$$VIEWER_CMD" \
+	  --command " $(VIEWER) \
+	    --result $(LOGDIR)/result.xml \
+	    --coverage $(LOGDIR)/coverage.xml \
+	    --property $(LOGDIR)/property.xml \
+	    --srcdir $(SRCDIR) \
+	    --goto $(HARNESS_GOTO).goto \
+	    --reportdir $(PROOFDIR)/report \
+	    --config $(PROOFDIR)/cbmc-viewer.json" \
 	  --inputs $^ \
-	  --outputs $(PROOFDIR)/html \
+	  --outputs $(PROOFDIR)/report \
 	  --pipeline-name "$(PROOF_UID)" \
-	  --ci-stage report \
 	  --stdout-file $(LOGDIR)/viewer-log.txt \
+	  --ci-stage report \
 	  --description "$(PROOF_UID): generating report"
 
-
-# Caution: run make-source before running property and coverage checking
-# The current make-source script removes the goto binary
-$(LOGDIR)/source.json:
-	mkdir -p $(dir $@)
-	$(RM) -r $(GOTODIR)
-	$(MAKE_SOURCE) --srcdir $(SRCDIR) --wkdir $(PROOFDIR) > $@
-	$(RM) -r $(GOTODIR)
-
-define VIEWER2_CMD
-  $(VIEWER2) \
-    --result $(LOGDIR)/result.xml \
-    --coverage $(LOGDIR)/coverage.xml \
-    --property $(LOGDIR)/property.xml \
-    --srcdir $(SRCDIR) \
-    --goto $(HARNESS_GOTO).goto \
-    --reportdir $(PROOFDIR)/report \
-    --config $(PROOFDIR)/cbmc-viewer.json
-endef
-export VIEWER2_CMD
-
-# Omit logs/source.json from report generation until make-sources
-# works correctly with Makefiles that invoke the compiler with
-# mutliple source files at once.
-$(PROOFDIR)/report: $(LOGDIR)/result.xml $(LOGDIR)/property.xml $(LOGDIR)/coverage.xml
+$(PROOFDIR)/report-no-coverage: $(LOGDIR)/result.xml $(LOGDIR)/property.xml
 	$(LITANI) add-job \
-	  --command "$$VIEWER2_CMD" \
+	  --command " $(VIEWER) \
```

Comment:
This seems like a lot of duplicated code just to change one flag. I think there should just be one recipe for report, and the coverage flag should be defined out if the user asks for it. Something like this (not tested):
```
# early in the file
VIEWER_COVER_FLAG ?= "--coverage $(LOGDIR)/coverage.xml"

report-no-coverage:
  $(MAKE) VIEWER_COVER_FLAG="" _report

# rule for report
# ...
        --command " $(VIEWER) \
	    --result $(LOGDIR)/result.xml \
	    $(VIEWER_COV_FLAG) \
	    --property $(LOGDIR)/property.xml \
```

Reply:
(and have a different variable for the output directory if it also needs to be different, etc)

Reply:
I've made these changes.